src/plugins/intel_cpu/src/nodes/kernels/dft_uni_kernel.cpp

```diff
@@ -0,0 +1,272 @@
+
+#define GET_OFF_DFT(field) offsetof(jit_args_dft, field)
+#define GET_OFF_FFT(field) offsetof(jit_args_fft, field)
+
+namespace ov {
+namespace intel_cpu {
+
+template <cpu::x64::cpu_isa_t isa>
+jit_uni_dft_kernel_f32<isa>::jit_uni_dft_kernel_f32(jit_dft_config_params jcp) : jcp_(jcp), jit_uni_dft_kernel(), jit_generator() {}
+
+template <cpu::x64::cpu_isa_t isa>
+void jit_uni_dft_kernel_f32<isa>::create_ker() {
+    jit_generator::create_kernel();
+    ker_ = (decltype(ker_))jit_ker();
+}
+
+template <cpu::x64::cpu_isa_t isa>
+void jit_uni_dft_kernel_f32<isa>::generate() {
+    this->preamble();
+
+    mov(reg_src, ptr[reg_params + GET_OFF_DFT(src)]);
+    mov(reg_dst, ptr[reg_params + GET_OFF_DFT(dst)]);
+    mov(reg_twiddles, ptr[reg_params + GET_OFF_DFT(twiddles)]);
+    mov(reg_work_amount, ptr[reg_params + GET_OFF_DFT(work_amount)]);
+    mov(reg_index, ptr[reg_params + GET_OFF_DFT(index)]);
+
+    Xbyak::Label main_loop_label;
+    Xbyak::Label main_loop_end_label;
+    Xbyak::Label tail_loop_label;
+    Xbyak::Label tail_loop_end_label;
+
+    mov(aux_reg_work_amount, reg_work_amount);
+    uni_vpxor(vmm_sum, vmm_sum, vmm_sum);
+
+    int step = vlen / 8;
+
+    L(main_loop_label);
+    {
+        cmp(aux_reg_work_amount, step);
+        jl(main_loop_end_label, T_NEAR);
+
+        uni_vpshufd(vmm_data, ptr[reg_src], 0b01000001);
+        uni_vpshufd(vmm_twiddles, ptr[reg_twiddles], 0b01000100);
+        uni_vfmadd231ps(vmm_sum, vmm_data, vmm_twiddles);
+
+        uni_vpshufd(vmm_data, ptr[reg_src], 0b11101011);
+        uni_vpshufd(vmm_twiddles, ptr[reg_twiddles], 0b11101110);
+        uni_vfmadd231ps(vmm_sum, vmm_data, vmm_twiddles);
+
+        add(reg_twiddles, 2 * step * sizeof(float));
+        add(reg_src, 2 * step * sizeof(float));
+
+        sub(aux_reg_work_amount, step);
+        jmp(main_loop_label, T_NEAR);
+    }
+    L(main_loop_end_label);
+
+    if (mayiuse(cpu::x64::avx512_core)) {
+        Xbyak::Zmm zmm_sum = Xbyak::Zmm(vmm_sum.getIdx());
+        Xbyak::Ymm ymm_sum = Xbyak::Ymm(vmm_sum.getIdx());
+        Xbyak::Ymm ymm_sum_2 = Xbyak::Ymm(vmm_sum_2.getIdx());
+
+        vextractf64x4(ymm_sum_2, zmm_sum, 1);
+        vaddps(ymm_sum, ymm_sum, ymm_sum_2);
+    }
+    if (mayiuse(cpu::x64::avx2)) {
+        Xbyak::Ymm ymm_sum = Xbyak::Ymm(vmm_sum.getIdx());
+
+        vextractf128(xmm_sum_2, ymm_sum, 1);
+        vaddps(xmm_sum, xmm_sum, xmm_sum_2);
+    }
+
+    L(tail_loop_label);
+    {
+        cmp(aux_reg_work_amount, 1);
+        jl(tail_loop_end_label, T_NEAR);
+
+        uni_vpshufd(xmm_data, ptr[reg_src], 0b01000001);
+        uni_vpshufd(xmm_twiddles, ptr[reg_twiddles], 0b01000100);
+        uni_vfmadd231ps(xmm_sum, xmm_data, xmm_twiddles);
+
+        add(reg_twiddles, 2 * sizeof(float));
+        add(reg_src, 2 * sizeof(float));
+
+        sub(aux_reg_work_amount, 1);
+        jmp(tail_loop_label, T_NEAR);
+    }
+    L(tail_loop_end_label);
+
+    vmovhlps(xmm_sum_2, xmm_sum_2, xmm_sum);
```

Comment:
Was it tested on the SSE41? Looks like it should not work there as well as number of instructions below.

Reply:
I added appropriate instructions and checked with ONEDNN_MAX_CPU_ISA=sse41; all tests passed

Reply:
It's better to use this [emulator](https://github.com/openvinotoolkit/openvino/wiki/CPUPluginIntelSoftwareDevelopmentEmulator).